initrd/etc/functions

```diff
@@ -284,6 +284,19 @@ replace_config() {
 	rm -f ${CONFIG_FILE}.tmp
 }
 
+# Generate secret value using first 20 chars of ROM SHA256 hash
+secret_from_rom_hash() {
+	local ROM_IMAGE="/tmp/coreboot-notpm.rom"
+
+	echo -e "\nTPM not detected; measuring ROM directly\n" 1>&2
+	# use a previously-copied image if it exists
+	if [ -f ${ROM_IMAGE} ]; then
+		sha256sum ${ROM_IMAGE} | cut -f1 -d ' ' | cut -c 1-20 | tr -d '\n'
```

Comment:
I thought Nitrokey firmware updated the size of HOTP secret. Why is this still truncated?
Will search for old issue on nitrokey side if that doesn't ring a bell.

Reply:
Good question.  If it was updated, maybe it is for compatibility with older firmware?  If so, and we can get the firmware version, I could have it only truncate if needed.  If you can find that issue to pinpoint the change let me know, I'll look too.

Reply:
@JonathonHall-Purism https://github.com/osresearch/heads/pull/836#issuecomment-694261654

Reply:
https://github.com/osresearch/heads/pull/493#pullrequestreview-189586537

Reply:
@JonathonHall-Purism https://github.com/osresearch/heads/pull/493#issuecomment-487251001

Reply:
Seems like it was fixed in firmware in 2019?

Reply:
> @JonathonHall-Purism [#493 (comment)](https://github.com/osresearch/heads/pull/493#issuecomment-487251001)

Seems like it was fixed in firmware in 2019? https://github.com/Nitrokey/nitrokey-storage-firmware/commit/ec5c0c13076a2064da330ee9bb6206dccaadee71

Reply:
@tlaurion Firmware supports it per your notes, but `hotp` from tpmtotp still requires exactly 20 bytes: https://github.com/osresearch/tpmtotp/blob/master/hotp.c#L26

The secret does not have to be printable, so I can use 20 binary bytes from the hash and get 160 bits of entropy, up from 80 currently (20 hex chars).

tpmtotp can be extended but I feel like this is going down a rabbit hole to go from 160 bits to 256 bits.  Can we track that for an improvement later?

Reply:
@tlaurion :point_up: Thread on secret truncation